diff -Naur linux-source-2.6.32-openvz-042stab084.14-amd64/drivers/scsi/mvsas/mv_defs.h linux-source-2.6.32-openvz-042stab084.14-amd64-ocz/drivers/scsi/mvsas/mv_defs.h
--- linux-source-2.6.32-openvz-042stab084.14-amd64/drivers/scsi/mvsas/mv_defs.h	2009-12-03 04:51:21.000000000 +0100
+++ linux-source-2.6.32-openvz-042stab084.14-amd64-ocz/drivers/scsi/mvsas/mv_defs.h	2014-01-13 01:49:39.000000000 +0100
@@ -33,6 +33,7 @@
 	chip_6440,
 	chip_6485,
 	chip_9480,
+	chip_9485,
 	chip_9180,
 	chip_1300,
 	chip_1320
diff -Naur linux-source-2.6.32-openvz-042stab084.14-amd64/drivers/scsi/mvsas/mv_init.c linux-source-2.6.32-openvz-042stab084.14-amd64-ocz/drivers/scsi/mvsas/mv_init.c
--- linux-source-2.6.32-openvz-042stab084.14-amd64/drivers/scsi/mvsas/mv_init.c	2013-09-14 10:53:01.000000000 +0200
+++ linux-source-2.6.32-openvz-042stab084.14-amd64-ocz/drivers/scsi/mvsas/mv_init.c	2014-01-13 01:49:59.000000000 +0100
@@ -32,6 +32,7 @@
 	[chip_6485] =	{ 1, 8, 0x800, 33, 32, 10, &mvs_64xx_dispatch, },
 	[chip_9180] =	{ 2, 4, 0x800, 17, 64,  9, &mvs_94xx_dispatch, },
 	[chip_9480] =	{ 2, 4, 0x800, 17, 64,  9, &mvs_94xx_dispatch, },
+	[chip_9485] =   { 2, 4, 0x800, 17, 64, 11, &mvs_94xx_dispatch, },
 	[chip_1300] =	{ 1, 4, 0x400, 17, 16,  9, &mvs_64xx_dispatch, },
 	[chip_1320] =	{ 2, 4, 0x800, 17, 64,  9, &mvs_94xx_dispatch, },
 };
@@ -657,7 +658,16 @@
 	{ PCI_VDEVICE(ARECA, PCI_DEVICE_ID_ARECA_1300), chip_1300 },
 	{ PCI_VDEVICE(ARECA, PCI_DEVICE_ID_ARECA_1320), chip_1320 },
 	{ PCI_VDEVICE(ADAPTEC2, 0x0450), chip_6440 },
-
+	{ PCI_VDEVICE(OCZ, 0x1021), chip_9485}, /* OCZ RevoDrive3 */
+	{ PCI_VDEVICE(OCZ, 0x1022), chip_9485}, /* OCZ RevoDrive3/zDriveR4 (exact model unknown) */
+	{ PCI_VDEVICE(OCZ, 0x1040), chip_9485}, /* OCZ RevoDrive3/zDriveR4 (exact model unknown) */
+	{ PCI_VDEVICE(OCZ, 0x1041), chip_9485}, /* OCZ RevoDrive3/zDriveR4 (exact model unknown) */
+	{ PCI_VDEVICE(OCZ, 0x1042), chip_9485}, /* OCZ RevoDrive3/zDriveR4 (exact model unknown) */
+	{ PCI_VDEVICE(OCZ, 0x1043), chip_9485}, /* OCZ RevoDrive3/zDriveR4 (exact model unknown) */
+	{ PCI_VDEVICE(OCZ, 0x1044), chip_9485}, /* OCZ RevoDrive3/zDriveR4 (exact model unknown) */
+	{ PCI_VDEVICE(OCZ, 0x1080), chip_9485}, /* OCZ RevoDrive3/zDriveR4 (exact model unknown) */
+	{ PCI_VDEVICE(OCZ, 0x1083), chip_9485}, /* OCZ RevoDrive3/zDriveR4 (exact model unknown) */
+	{ PCI_VDEVICE(OCZ, 0x1084), chip_9485}, /* OCZ RevoDrive3/zDriveR4 (exact model unknown) */
 	{ }	/* terminate list */
 };
diff -Naur linux-source-2.6.32-openvz-042stab084.14-amd64/include/linux/pci_ids.h linux-source-2.6.32-openvz-042stab084.14-amd64-ocz/include/linux/pci_ids.h
--- linux-source-2.6.32-openvz-042stab084.14-amd64/include/linux/pci_ids.h	2013-12-16 15:00:53.000000000 +0100
+++ linux-source-2.6.32-openvz-042stab084.14-amd64-ocz/include/linux/pci_ids.h	2014-01-13 01:24:38.000000000 +0100
@@ -2837,3 +2837,5 @@
 
 #define PCI_VENDOR_ID_XEN	0x5853
 #define PCI_DEVICE_ID_XEN_PLATFORM	0x0001
+
+#define PCI_VENDOR_ID_OCZ		0x1b85
